src/collection-view.js

```diff
@@ -46,33 +50,42 @@ Marionette.CollectionView = Marionette.View.extend({
   },
 
   endBuffering: function() {
+    // Only trigger attach if already shown and attached, otherwise Region#show() handles this.
+    var canTriggerAttach = this._isShown && Marionette.isNodeAttached(this.el);
+    var nestedViews;
+
     this.isBuffering = false;
-    this._triggerBeforeShowBufferedChildren();
 
-    this.attachBuffer(this);
+    if (this._isShown) {
```

Comment:
So, I put a lot of thought/research into reasoning this one out.  It seems like the previous code was written such that a collection view does _not_ behave like a region (firing lifecycle events on child views) _unless_ it's situated within a Region, for one.  For two, if a view lifecycle event is the Region's responsibility (`attach`, `before:show`, `show) and we _don't_ have that check, view lifecycle events could be triggered twice on child views.  One question to table is that perhaps a collection view _should_ behave like a region even when it's not within a region, but then we would have to solve the double-triggered child view events.